src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -856,6 +860,29 @@ public function handleLabTable(LabTablePacket $packet) : bool{
 		return false; //TODO
 	}
 
+	public function handleLecternUpdate(LecternUpdatePacket $packet) : bool{
+		//Drop book is handled on server side with an interact event.
+		//Total pages is gathered from book.
+		$pos = $packet->blockPosition;
+		$chunkX = $pos->getX() >> Chunk::COORD_BIT_SIZE;
+		$chunkZ = $pos->getZ() >> Chunk::COORD_BIT_SIZE;
+		$world = $this->player->getWorld();
+		if(!$world->isChunkLoaded($chunkX, $chunkZ) || !$world->isChunkGenerated($chunkX, $chunkZ) || $world->isChunkLocked($chunkX, $chunkZ)){
+			return false;
+		}
+		$lectern = $world->getBlockAt($pos->getX(), $pos->getY(), $pos->getZ());
+		if($lectern instanceof Lectern){
+			if($this->player->getPosition()->distance($lectern->getPosition()) <= 15){
```

Comment:
This check can be combined with the one above.

In addition, this should probably use `Player->canInteract()`.